request/src/main/java/com/vimeo/networking2/VimeoApiClient.kt

```diff
@@ -422,6 +410,147 @@ interface VimeoApiClient {
      */
     fun deleteConnectedApp(type: ConnectedAppType, callback: VimeoCallback<Unit>): VimeoRequest
 
+    /**
+     * Create a folder that will be used to organize videos.
+     *
+     * @param uri The URI of the user's projects connection.
+     * @param name The name of the folder.
+     * @param privacy The privacy of the folder.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun createFolder(
+        uri: String,
+        name: String,
+        privacy: FolderViewPrivacyType,
+        callback: VimeoCallback<Folder>
+    ): VimeoRequest
+
+    /**
+     * Create a folder that will be used to organize videos.
+     *
+     * @param user The user whose projects connection will be used for the request.
+     * @param name The name of the folder.
+     * @param privacy The privacy of the folder.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun createFolder(
+        user: User,
+        name: String,
+        privacy: FolderViewPrivacyType,
+        callback: VimeoCallback<Folder>
+    ): VimeoRequest
+
+    /**
+     * Delete a folder.
+     *
+     * @param folder The folder being deleted, URI should not be null or empty.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun deleteFolder(
+        folder: Folder,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest
+
+    /**
+     * Edit a folder's information.
+     *
+     * @param uri The URI of the user's projects connection.
+     * @param name The name of the folder.
+     * @param privacy The privacy of the folder.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun editFolder(
+        uri: String,
+        name: String,
+        privacy: FolderViewPrivacyType,
+        callback: VimeoCallback<Folder>
```

Comment:
The api also supports slack related parameters here. Maybe we should add them in now so we are ready to support this later? Web currently exposes it.

https://github.vimeows.com/Vimeo-API/design/blob/master/docs/3.4.2/apiblueprint/resources/Folders/Essentials.apib